monitoring-engine/src/main/java/io/jenkins/plugins/remotingopentelemetry/engine/EngineConfiguration.java

```diff
@@ -16,11 +16,32 @@
     private final int exporterTimeoutMillis = 30_000;
 
     @Nonnull
-    private String nodeName = "";
+    private final String nodeName;
+
+    @Nonnull
```

Comment:
FTR it is possible to make fields , return values and parameters nonnull by default on the package level. Example: https://github.com/oleg-nenashev/FaaScinator/blob/main/function/3_service/src/main/java/io/faascinator/service/util/package-info.java . It might be more convenient though it makes things less obvious in Javadoc browsers